### PR TITLE
Add fee getter

### PR DESCRIPTION
**Pull request by zancas**

### CODE REVIEW

File: zcash_primitives/src/transaction/builder.rs [hunk 1]
```diff
@@ -402,17 +402,16 @@ impl<'a, P: consensus::Parameters, R: RngCore + CryptoRng> Builder<'a, P, R> {
...
-        prover: &impl TxProver,
+    /// This fee is a function of the spends and outputs that
+    /// have been added to the builder and the FeeRule, that
+    /// is supplied as this methods only free parameter.
+    pub fn get_fee<FR: FeeRule>(
+        &self,
         fee_rule: &FR,
-    ) -> Result<(Transaction, SaplingMetadata), Error<FR::Error>> {
-        let fee = fee_rule
+    ) -> Result<Amount, Error<<FR as super::fees::FeeRule>::Error>> {
```
Comment (nuttycom) [inline zcash_primitives/src/transaction/builder.rs:413]:
```suggestion
    ) -> Result<Amount, Error<FR::Error>> {
```

File: zcash_primitives/src/transaction/builder.rs [hunk 2]
```diff
@@ -402,17 +402,16 @@ impl<'a, P: consensus::Parameters, R: RngCore + CryptoRng> Builder<'a, P, R> {
             .sum::<Option<_>>()
             .ok_or(BalanceError::Overflow)
     }
-
-    /// Builds a transaction from the configured spends and outputs.
+    /// Reports the implied fee
```
Comment (daira) [inline zcash_primitives/src/transaction/builder.rs:405]:
```suggestion
    /// Reports the calculated fee for the given [`FeeRule`].
```

File: zcash_primitives/src/transaction/builder.rs [hunk 3]
```diff
@@ -402,17 +402,16 @@ impl<'a, P: consensus::Parameters, R: RngCore + CryptoRng> Builder<'a, P, R> {
...
     ///
-    /// Upon success, returns a tuple containing the final transaction, and the
-    /// [`SaplingMetadata`] generated during the build process.
-    pub fn build<FR: FeeRule>(
-        self,
-        prover: &impl TxProver,
+    /// This fee is a function of the spends and outputs that
+    /// have been added to the builder and the FeeRule, that
+    /// is supplied as this methods only free parameter.
```
Comment (daira) [inline zcash_primitives/src/transaction/builder.rs:409]:
```suggestion
    /// have been added to the builder and of [`fee_rule`].
```

File: zcash_primitives/src/transaction/builder.rs (hunk 2 above)
Comment (nuttycom) [inline zcash_primitives/src/transaction/builder.rs:405]:
```suggestion

    /// Reports the implied fee
```

**Review by nuttycom — CHANGES_REQUESTED** [review]
A few minor changes, I'll add a cleanup commit.

**Review by daira — APPROVED** [review]
utACK

**Review by zancas — COMMENTED** [review]
I squashed into a single commit.  What do y'all think?

File: zcash_primitives/src/transaction/builder.rs [hunk 4]
```diff
@@ -403,16 +403,12 @@ impl<'a, P: consensus::Parameters, R: RngCore + CryptoRng> Builder<'a, P, R> {
...
-    pub fn build<FR: FeeRule>(
-        self,
-        prover: &impl TxProver,
-        fee_rule: &FR,
-    ) -> Result<(Transaction, SaplingMetadata), Error<FR::Error>> {
-        let fee = fee_rule
+    /// This fee is a function of the spends and outputs that have been added to the builder,
+    /// pursuant to the specified [`FeeRule`].
+    pub fn get_fee<FR: FeeRule>(&self, fee_rule: &FR) -> Result<Amount, Error<FR::Error>> {
+        fee_rule
```
Comment (daira) [inline zcash_primitives/src/transaction/builder.rs:411]: Please run `cargo fmt`, I think that hasn't been done.

**Review by daira — APPROVED** [review]
utACK

File: zcash_primitives/src/transaction/builder.rs (hunk 4 above)
Comment (nuttycom) [reply zcash_primitives/src/transaction/builder.rs:411]: The `rustfmt` check passed, so I think this is fine.

**Pull request merged**